python/k8s.py

```diff
@@ -488,28 +488,80 @@ def install_ingress_nginx(version):
+@option('--grafana-admin-password', default='grafana', help="Grafana admin password")
+def install_kube_prometheus_stack(version, alertmanager, pushgateway, coredns, kubedns,
+                                  kube_scheduler, kube_controller_manager,
+                                  prometheus_retention, prometheus_persistence_size,
+                                  grafana_host, grafana_persistence_size, grafana_admin_password):
+    """Install a kube-prometheus-stack instance in the current cluster"""
     call(['helm', 'repo', 'add', 'prometheus-community', 'https://prometheus-community.github.io/helm-charts'])
+    call(['helm', 'repo', 'update'])
     call([
         'helm', '--kube-context', config.kubectl.context,
-        'upgrade', '--install', '--create-namespace', '--wait', 'prometheus', 'prometheus-community/prometheus',
+        'upgrade', '--install', '--create-namespace', '--wait', 'kube-prometheus-stack',
+        'prometheus-community/kube-prometheus-stack',
         '--namespace', 'monitoring',
         '--version', version,
         '--set', 'alertmanager.enabled=' + str(alertmanager).lower(),
         '--set', 'pushgateway.enabled=' + str(pushgateway).lower(),
-        '--set', 'server.strategy.type=Recreate',
-        '--set', 'server.retention=' + retention,
-        '--set', 'nodeExporter.hostRootfs=' + str(not(config.k8s.distribution == "docker-desktop")).lower(),
-        '--set', 'server.persistentVolume.size=' + persistent_volume_size,
+        '--set', 'coreDns.enabled=' + str(coredns).lower(),
+        '--set', 'kubeDns.enabled=' + str(kubedns).lower(),
+        '--set', 'kubeScheduler.enabled=' + str(kube_scheduler).lower(),
+        '--set', 'kubeControllerManager.enabled=' + str(kube_controller_manager).lower(),
+        '--set', 'prometheus.prometheusSpec.retention=' + prometheus_retention,
+        '--set', 'prometheus.prometheusSpec.persistentVolume.size=' + prometheus_persistence_size,
+        '--set', 'prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues=false',
+        '--set', 'prometheus-node-exporter.hostRootFsMount=' + str(not (config.k8s.distribution == "docker-desktop")).lower(),
+        '--set', 'grafana.ingress.enable=true',
+        '--set', 'grafana.ingress.hosts[0]=' + str(grafana_host),
+        '--set', 'grafana.adminPassword=' + str(grafana_admin_password),
+        '--set', 'grafana.persistence.enabled=true',
+        '--set', 'grafana.persistence.size=' + grafana_persistence_size,
+        '--set', 'grafana.deploymentStrategy.type=Recreate',
     ])  # yapf: disable
 
 
+@k8s.command(flowdepends=['k8s.create-cluster'])
+@option('--version', default='v0.50.0', help="The version of prometheus operator CRDs to install")
+def install_prometheus_operator_crds(version):
+    """Install prometheus operator CRDs in the current cluster"""
+    config.kubectl.output(['apply', '-f', 'https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/'
```

Comment:
create a variable with the base url and reuse it after that

~~~python
base_url = f'https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/{version}/example/prometheus-operator-crd/'
config.kubectl.output(['apply', '-f', f'{base_url}/monitoring.coreos.com_alertmanagers.yaml')
~~~

maybe even use a for loop to avoid duplication `config.kubectl.output(['apply', '-f', …`